examples/beacon-api-sidecar-fetcher/src/main.rs

```diff
@@ -0,0 +1,235 @@
+
+    #[error("Data parsing error: {0}")]
+    DeserializationError(String),
+
+    #[error("{0} Error: {1}")]
+    UnknownError(u16, String),
+}
+
+#[derive(Debug, Clone)]
+pub struct MinedSidecar {
+    tx_hash: TxHash,
+    block_hash: BlockHash,
+    block_number: BlockNumber,
+    sidecar: BlobTransactionSidecar,
+}
+
+#[derive(Debug)]
+pub struct MinedSidecarStream<St, P>
+where
+    St: Stream<Item = CanonStateNotification> + Send + Unpin + 'static,
+{
+    events: St,
+    pool: P,
+    client: reqwest::Client,
+    pending_requests:
+        FuturesUnordered<Pin<Box<dyn Future<Output = Result<Vec<MinedSidecar>, SideCarError>>>>>, /* TODO make vec */
+    queued_actions: VecDeque<MinedSidecar>, /* Buffer for
+                                             * ready items */
+}
+
+impl<St, P> Stream for MinedSidecarStream<St, P>
+where
+    St: Stream<Item = CanonStateNotification> + Send + Unpin + 'static,
+    P: TransactionPoolExt + Unpin + 'static,
+{
+    type Item = Result<MinedSidecar, SideCarError>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let this: &mut MinedSidecarStream<St, P> = self.get_mut();
+
+        // return any buffered result
+        if let Some(mined_sidecar) = this.queued_actions.pop_front() {
+            return Poll::Ready(Some(Ok(mined_sidecar)));
+        }
+
+        // Check if any pending reqwests are ready and append to buffer
+        while let Poll::Ready(Some(pending_result)) = this.pending_requests.poll_next_unpin(cx) {
+            match pending_result {
+                Ok(mined_sidecars) => {
+                    for sidecar in mined_sidecars {
+                        this.queued_actions.push_back(sidecar);
+                    }
+
+                    // Try returning the next available sidecar if any
+                    if let Some(sidecar) = this.queued_actions.pop_front() {
+                        return Poll::Ready(Some(Ok(sidecar)));
+                    }
+                }
+                Err(err) => return Poll::Ready(Some(Err(err))),
+            }
+        }
+
+        loop {
+            match this.events.poll_next_unpin(cx) {
+                Poll::Ready(Some(notification)) => {
+                    let notification_clone = notification.clone();
+                    let mut all_blobs_available = true;
+                    let mut actions_to_queue: Vec<MinedSidecar> = Vec::new();
+
+                    let txs: Vec<_> = notification
+                        .tip()
+                        .transactions()
+                        .filter(|tx| tx.is_eip4844())
+                        .map(|tx| {
+                            // Clone minimal data necessary for later processing
+                            (tx.hash, tx.blob_versioned_hashes().unwrap().len())
+                        })
+                        .collect();
+
+                    for (tx_hash, _) in &txs {
+                        match this.pool.get_blob(*tx_hash) {
```

Comment:
I think we want

https://github.com/paradigmxyz/reth/blob/250da4e449d1872625fa6d8b0a943ceb0f6b8379/crates/transaction-pool/src/traits.rs#L378-L385

here

Reply:
Since we have moved to this, I have change the logic to zip the txs alongside the sidercars as they should be in the same exact order.

Just need to work out the error handling logic here.